### PR TITLE
Fix permissions so a supervisor may see files of exercises he/she can assign

### PR DESCRIPTION
**Pull request by krulis-martin**

### CODE REVIEW

File: app/V1Module/security/Policies/UploadedFilePermissionPolicy.php [hunk 1]
```diff
@@ -58,6 +58,30 @@ public function isAuthorOfSupplementaryFileExercises(Identity $identity, Uploade
...
+
+    $user = $identity->getUserData();
+    if ($user === null) {
+      return false;
+    }
+
+    foreach ($file->getExercises() as $exercise) {
+      foreach ($exercise->getGroups() as $group) {
+        if ($group->isAdminOrSupervisorOfSubgroup($user)) {
+          return true;  // The user can assign one of the corresponding exercises in hir group.
```
Comment (Teyras) [inline app/V1Module/security/Policies/UploadedFilePermissionPolicy.php]: Is `hir` a typo or a gender-neutral pronoun?

File: app/V1Module/security/Policies/UploadedFilePermissionPolicy.php [hunk 2]
```diff
@@ -58,6 +58,30 @@ public function isAuthorOfSupplementaryFileExercises(Identity $identity, Uploade
     return FALSE;
   }
 
+
+  public function isInGroupUserSupervises(Identity $identity, UploadedFile $file)
```
Comment (Teyras) [inline app/V1Module/security/Policies/UploadedFilePermissionPolicy.php]: I understand that the name is long enough as is, but it should probably reflect the fact that this assertion only works for supplementary (test) files.

File: app/V1Module/security/Policies/UploadedFilePermissionPolicy.php (hunk 1 above)
Comment (krulis-martin) [reply app/V1Module/security/Policies/UploadedFilePermissionPolicy.php]: The latter.

File: app/V1Module/security/Policies/UploadedFilePermissionPolicy.php (hunk 2 above)
Comment (krulis-martin) [reply app/V1Module/security/Policies/UploadedFilePermissionPolicy.php]: Any suggestions?

Comment (Teyras) [reply app/V1Module/security/Policies/UploadedFilePermissionPolicy.php]: isSupplementaryInGroupUserSupervises? That's awfully long. Is it necessary for the method to only work on supplementary files? Don't additional files have a list of exercises too?

**Pull request merged**